src/field_theory/normal.lean

```diff
@@ -294,4 +294,23 @@ lemma alg_equiv.restrict_normal_hom_surjective [normal F K] [normal F E] :
   function.surjective (alg_equiv.restrict_normal_hom K : (E ≃ₐ[F] E) → (K ≃ₐ[F] K)) :=
 λ χ, ⟨χ.lift_normal E, χ.restrict_lift_normal E⟩
 
+variables (F) (K) (E)
+
+lemma is_solvable_of_is_scalar_tower [normal F K] [h1 : is_solvable (K ≃ₐ[F] K)]
+  [h2 : is_solvable (E ≃ₐ[K] E)] : is_solvable (E ≃ₐ[F] E) :=
+begin
+  let f : (E ≃ₐ[K] E) →* (E ≃ₐ[F] E) :=
+  { to_fun := λ ϕ, alg_equiv.of_alg_hom (ϕ.to_alg_hom.restrict_scalars F)
+      (ϕ.symm.to_alg_hom.restrict_scalars F)
+      (alg_hom.ext (λ x, ϕ.apply_symm_apply x))
+      (alg_hom.ext (λ x, ϕ.symm_apply_apply x)),
+    map_one' := alg_equiv.ext (λ _, rfl),
+    map_mul' := λ _ _, alg_equiv.ext (λ _, rfl) },
+  let g : (E ≃ₐ[F] E) →* (K ≃ₐ[F] K) := alg_equiv.restrict_normal_hom K,
+  have key : g.ker ≤ f.range := λ ϕ hϕ, ⟨alg_equiv.mk ϕ ϕ.symm ϕ.symm_apply_apply
+    ϕ.apply_symm_apply ϕ.map_mul ϕ.map_add (λ x, eq.trans (ϕ.restrict_normal_commutes K x).symm
+    (congr_arg (algebra_map K E) (alg_equiv.ext_iff.mp hϕ x))), alg_equiv.ext (λ _, rfl)⟩,
+  exact solvable_of_ker_le_range f g key,
```

Comment:
Here's another way to write the proof. Can you tell that I like "backwards" reasoning? :)

The main point to me is replacing `alg_equiv.mk` with `{commutes' := _, .. ϕ}`. (This could even be packaged into its own construction if it comes up more, something like `alg_equiv.expand_scalars`.)
```suggestion
  let f : (E ≃ₐ[K] E) →* (E ≃ₐ[F] E) :=
  { to_fun := λ ϕ, alg_equiv.of_alg_hom (ϕ.to_alg_hom.restrict_scalars F)
      (ϕ.symm.to_alg_hom.restrict_scalars F)
      (alg_hom.ext (λ x, ϕ.apply_symm_apply x))
      (alg_hom.ext (λ x, ϕ.symm_apply_apply x)),
    map_one' := alg_equiv.ext (λ _, rfl),
    map_mul' := λ _ _, alg_equiv.ext (λ _, rfl) },
  let g : (E ≃ₐ[F] E) →* (K ≃ₐ[F] K) := alg_equiv.restrict_normal_hom K,
  refine solvable_of_ker_le_range f g (λ ϕ hϕ, ⟨{commutes' := _, .. ϕ}, alg_equiv.ext (λ _, rfl)⟩),
  intro x,
  calc ϕ (algebra_map K E x)
      = algebra_map K E (alg_equiv.restrict_normal ϕ K x) : (ϕ.restrict_normal_commutes K x).symm
  ... = algebra_map K E x : congr_arg (algebra_map K E) (alg_equiv.ext_iff.mp hϕ x),
```

Reply:
Nice. I didn't use the calc proof at the end, but if you want me to then I can.